src/Player.cpp

```diff
@@ -564,11 +564,11 @@ JResult CPlayer::Magic( CLink<CItem> *pLink )
     // Quaff and Magic are similar but magic is multi-use
     CItem *pItem = pLink->m_lpData;
     CLink<CEffect> *plEffect = pItem->m_id->m_llEffects->GetHead();
-    JResult retval = DoEffects( plEffect );
+    JResult retval = DoEffects( plEffect, pItem->m_dwFlags );
     return retval;
 }
 
-JResult CPlayer::DoEffects( CLink<CEffect> *plEffect )
+JResult CPlayer::DoEffects( CLink<CEffect> *plEffect, int dwFlags )
```

Comment:
Please rename second parameter to dwItemFlags, to distinguish from the ones in the Effect